internal/service/encryptionatrest/resource_encryption_at_rest.go

```diff
@@ -216,19 +216,19 @@ func (r *encryptionAtRestRS) Create(ctx context.Context, req resource.CreateRequ
 	projectID := encryptionAtRestPlan.ProjectID.ValueString()
 	encryptionAtRestReq := &admin.EncryptionAtRest{}
 	if encryptionAtRestPlan.AwsKmsConfig != nil {
-		encryptionAtRestReq.AwsKms = newAtlasAwsKms(encryptionAtRestPlan.AwsKmsConfig)
+		encryptionAtRestReq.AwsKms = NewAtlasAwsKms(encryptionAtRestPlan.AwsKmsConfig)
 	}
 	if encryptionAtRestPlan.AzureKeyVaultConfig != nil {
-		encryptionAtRestReq.AzureKeyVault = newAtlasAzureKeyVault(encryptionAtRestPlan.AzureKeyVaultConfig)
+		encryptionAtRestReq.AzureKeyVault = NewAtlasAzureKeyVault(encryptionAtRestPlan.AzureKeyVaultConfig)
 	}
 	if encryptionAtRestPlan.GoogleCloudKmsConfig != nil {
```

Comment:
NewAtlasGcpKms is already checking if param is nil so i think the is condition is not needed, it will just asssign nil that is the zero value that it already had. 

 same for the others

Reply:
You are right! Changed